SDWebImage/Core/SDWebImagePrefetcher.m

```diff
@@ -142,13 +145,13 @@ - (void)startPrefetchWithToken:(SDWebImagePrefetchToken * _Nonnull)token {
                     [asyncOperation complete];
                 }];
                 NSAssert(operation != nil, @"Operation should not be nil, [SDWebImageManager loadImageWithURL:options:context:progress:completed:] break prefetch logic");
-                @synchronized (token) {
-                    [operations addPointer:(__bridge void *)operation];
-                }
+                SD_LOCK(token->_loadOperationsLock);
+                [token.loadOperations addPointer:(__bridge void *)operation];
+                SD_UNLOCK(token->_loadOperationsLock);
```

Comment:
We must use two different lock for these two NSPoiniterArray. One for `loadOperations` one for `prefetchOperations`.

Using the same lock (via `@synchronized`) may cause deadlock, when, for example, when the `cancel` called from user (the `@synchronized` entered 1), and this enter the SDWebImageManager's cancel completion block (the `@synchronized` wil entered 2), and wait forever.